ecs-cli/modules/cli/configure/migrate.go

```diff
@@ -0,0 +1,93 @@
+// Copyright 2015-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package configure
+
+import (
+	"html/template"
+	"io/ioutil"
+	"os"
+
+	"github.com/aws/amazon-ecs-cli/ecs-cli/modules/config"
+)
+
+func migrateWarning(cliConfig *config.CLIConfig) error {
+	var oldConfig string
+	dest, err := config.NewDefaultDestination()
+	if err != nil {
+		return err
+	}
+	dat, err := ioutil.ReadFile(config.ConfigFilePath(dest))
+	if err != nil {
+		return err
+	}
+	oldConfig = string(dat)
+
+	// format template
+	data := map[string]interface{}{
+		"OldConfig":    oldConfig,
+		"Cluster":      cliConfig.Cluster,
+		"Region":       cliConfig.Region,
+		"CFNStackName": "cfn-stack-name: " + cliConfig.CFNStackName,
+		"AWSAccessKey": cliConfig.AWSAccessKey,
+		"AWSSecretKey": cliConfig.AWSSecretKey,
+	}
+
+	if cliConfig.ComposeServiceNamePrefix != "" {
+		data["ComposeServiceNamePrefix"] = "compose-service-name-prefix: " + cliConfig.ComposeServiceNamePrefix
+	}
+
+	if cliConfig.CFNStackName != "" {
+		data["CFNStackName"] = "cfn-stack-name: " + cliConfig.CFNStackName
+	}
+
+	t := template.Must(template.New("message").Parse(messageTemplate))
+	if err := t.Execute(os.Stdout, data); err != nil {
+		return err
+	}
+
+	return nil
+
+}
+
+var messageTemplate = `
```

Comment:
Can you provide examples of the following cases?
* ini with default prefix --> yaml
* ini with empty prefix --> yaml
* ini with custom prefix --> yaml

Reply:
```
Old configuration file
-----------------------------------------------------
~/.ecs/config
[ecs]
cluster                     = ReleaseTestCluster
aws_profile                 = 
region                      = us-east-2
aws_access_key_id           = 
aws_secret_access_key       = 
compose-project-name-prefix = ecscompose-
compose-service-name-prefix = ecscompose-service-
cfn-stack-name-prefix       = amazon-ecs-cli-setup-



New configuration files
-----------------------------------------------------
~/.ecs/config
version: v1
default: default
cluster-configurations:
  default:
    cluster: ReleaseTestCluster
    region: us-east-2
    cfn-stack-name: amazon-ecs-cli-setup-ReleaseTestCluster
    compose-service-name-prefix: ecscompose-service-

~/.ecs/credentials
default: default
credentials:
  default:
    aws_access_key_id: 
    aws_secret_access_key:

[WARN] Please read the following changes carefully: http://docs.aws.amazon.com/AmazonECS/latest/developerguide/ECS_CLI_Configuration.html
- The --compose-project-name-prefix and --compose-service-name-prefix options are deprecated. You can specify your desired names with the --project-name option.
- The --cfn-stack-name-prefix option has been removed. To use an existing CloudFormation stack, please specify the full stack name; otherwise, the stack name defaults to amazon-ecs-cli-setup-<cluster_name>.
```

```
Old configuration file
-----------------------------------------------------
~/.ecs/config
[ecs]
cluster                     = ReleaseTestCluster
aws_profile                 = 
region                      = us-east-2
aws_access_key_id           = 
aws_secret_access_key       = 
compose-project-name-prefix = 
compose-service-name-prefix = 
cfn-stack-name-prefix       = 



New configuration files
-----------------------------------------------------
~/.ecs/config
version: v1
default: default
cluster-configurations:
  default:
    cluster: ReleaseTestCluster
    region: us-east-2
    cfn-stack-name: ReleaseTestCluster
    

~/.ecs/credentials
default: default
credentials:
  default:
    aws_access_key_id: 
    aws_secret_access_key: 

[WARN] Please read the following changes carefully: http://docs.aws.amazon.com/AmazonECS/latest/developerguide/ECS_CLI_Configuration.html
- The --compose-project-name-prefix and --compose-service-name-prefix options are deprecated. You can specify your desired names with the --project-name option.
- The --cfn-stack-name-prefix option has been removed. To use an existing CloudFormation stack, please specify the full stack name; otherwise, the stack name defaults to amazon-ecs-cli-setup-<cluster_name>.
```

```
Old configuration file
-----------------------------------------------------
~/.ecs/config
[ecs]
cluster                     = ReleaseTestCluster
aws_profile                 = 
region                      = us-east-2
aws_access_key_id           = 
aws_secret_access_key       = 
compose-project-name-prefix = custom-project-
compose-service-name-prefix = custom-service-
cfn-stack-name-prefix       = custom-cfn-



New configuration files
-----------------------------------------------------
~/.ecs/config
version: v1
default: default
cluster-configurations:
  default:
    cluster: ReleaseTestCluster
    region: us-east-2
    cfn-stack-name: custom-cfn-ReleaseTestCluster
    compose-service-name-prefix: custom-service-

~/.ecs/credentials
default: default
credentials:
  default:
    aws_access_key_id:
    aws_secret_access_key: 

[WARN] Please read the following changes carefully: http://docs.aws.amazon.com/AmazonECS/latest/developerguide/ECS_CLI_Configuration.html
- The --compose-project-name-prefix and --compose-service-name-prefix options are deprecated. You can specify your desired names with the --project-name option.
- The --cfn-stack-name-prefix option has been removed. To use an existing CloudFormation stack, please specify the full stack name; otherwise, the stack name defaults to amazon-ecs-cli-setup-<cluster_name>.
```

Reply:
Update: It now hides the AWS Secret Key like this:
```
Old configuration file
-----------------------------------------------------
~/.ecs/config
[ecs]
cluster                     = ReleaseTestCluster
aws_profile                 = 
region                      = us-east-2
aws_access_key_id           = AKIAJHZ3OZKX4ZPLR6MQ
aws_secret_access_key       = ********************kp
compose-project-name-prefix = ecscompose-
compose-service-name-prefix = ecscompose-service-
cfn-stack-name-prefix       = amazon-ecs-cli-setup-




New configuration files
-----------------------------------------------------
~/.ecs/config
version: v1
default: default
cluster-configurations:
  default:
    cluster: ReleaseTestCluster
    region: us-east-2
    compose-service-name-prefix: ecscompose-service-


~/.ecs/credentials
default: default
credentials:
  default:
    aws_access_key_id: AKIAJHZ3OZKX4ZPLR6MQ
    aws_secret_access_key: ********************kp
```
(Key pair shown in this example is not real).

Reply:
Its not real in the sense that its not a valid pair of creds that will allow anyone to access any resources in AWS.